### PR TITLE
Add docs for an empty `trusted_key_servers` config option

### LINKED ISSUE #7047: That you can have an empty trusted_key_servers config option is non-obvious
Sometimes you're running Synapse and not wanting to trust any other Synapse server as a keyserver (either for privacy reasons or you're running in a private federation).

In this case, it's not obvious that the way that you specify "No external key server please" is to set `trusted_key_servers` to an empty array `[]`. Commenting out the block will just default to `matrix.org`.

We should add a line in the config help for this option to state that contacting 3rd party trusted key servers can be disabled by setting `trusted_key_servers` to `[]`.

Synapse v1.11.1

### PR DESCRIPTION
**Pull request by dklimpel**
Fixes #7047

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

Signed-off-by: Dirk Klimpel dirk@klimpel.org

### CODE REVIEW

File: docs/usage/configuration/config_documentation.md [hunk 1]
```diff
@@ -2655,6 +2655,12 @@ is still supported for backwards-compatibility, but it is deprecated.
 warning on start-up. To suppress this warning, set
 `suppress_key_server_warning` to true.
 
+If the use of a trusted key server has to be deactivated, e.g. in a private
+federation or for privacy reasons, this can be realised by setting
+an empty array (`trusted_key_servers: []`). Then Synapse will request the keys
+directly from the communication partner. If Synapse does not get keys directly
+from the communication partner, the events of the other server will be rejected.
```
Comment (babolivier) [inline docs/usage/configuration/config_documentation.md:2662]: I think "communication partner" is a bit unclear here. I can assume you mean the keys will be fetched directly from the servers owning them, but I think that's a bit unclear to someone who's less familiar with Synapse.

**Review by babolivier — CHANGES_REQUESTED** [review]
Thanks! I think it just needs a bit of clarification.

File: docs/usage/configuration/config_documentation.md (hunk 1 above)
Comment (dklimpel) [reply docs/usage/configuration/config_documentation.md:2662]: Yes, what is a better wording? "federation partner", "server that owns the keys"? Other suggestions?

Comment (babolivier) [reply docs/usage/configuration/config_documentation.md:2662]: Yes I think "server that owns the keys" is clearer.

File: changelog.d/13999.doc [hunk 2]
```diff
@@ -0,0 +1 @@
+Add documentation for using an empty `trusted_key_servers` config option.
```
Comment (DMRobertson) [inline changelog.d/13999.doc:1]:
```suggestion
Explain how to disable the use of `trusted_key_servers`.
```

**Review by DMRobertson — APPROVED** [review]
It would be nice to explain why we might want to use a trusted signing key server at all. I don't have a reference for you. I'm guessing it's needed to e.g. verify messages from A.com that we received from B.com when A.com goes offline? However I suppose that can be another PR.

I think this wording is good enough and is a clear improvement. Thanks again Dirk!

**Pull request merged**